### PR TITLE
feat: optional to all ecs components

### PR DESCRIPTION
**Pull request by kuruk-mm**
After merge update https://github.com/decentraland/js-sdk-toolchain/pull/218 and https://github.com/decentraland/unity-renderer/pull/2809 with `@dcl/protocol@next`

### CODE REVIEW

File: ecs/components/AvatarAttach.proto [hunk 1]
```diff
@@ -12,5 +12,5 @@ enum AvatarAnchorPoint {
 
 message PBAvatarAttach {
   string avatar_id = 1;
-  AvatarAnchorPoint anchor_point_id = 2;
+  optional AvatarAnchorPoint anchor_point_id = 2;
```
Comment (menduz) [inline ecs/components/AvatarAttach.proto:15]: I think this one is not optional

File: ecs/components/AvatarShape.proto [hunk 2]
```diff
@@ -5,16 +5,16 @@ option (ecs_component_id) = 1080;
 import "common/Color3.proto";
 
 message PBAvatarShape {
-  string id = 1;
-  string name = 2;
-  string body_shape = 3;
-  Color3 skin_color = 4;
-  Color3 hair_color = 5;
-  Color3 eye_color = 6;
+  optional string id = 1;
```
Comment (menduz) [inline ecs/components/AvatarShape.proto:8]: I believe this one may be necessary?

File: ecs/components/BoxShape.proto [hunk 3]
```diff
@@ -4,8 +4,8 @@ import "common/id.proto";
 option (ecs_component_id) = 1013;
 
 message PBBoxShape {  
-  bool with_collisions = 1;
-  bool is_pointer_blocker = 2;
-  bool visible = 3;
+  optional bool with_collisions = 1; // default=true
+  optional bool is_pointer_blocker = 2; // default=true
+  optional bool visible = 3; // default=true
```
Comment (menduz) [inline ecs/components/BoxShape.proto:9]:
```suggestion
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool with_collisions = 1; // default=true
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool is_pointer_blocker = 2; // default=true
  // @deprecated use HiddenComponent instead https://github.com/decentraland/sdk/issues/353
  optional bool visible = 3; // default=true
```

File: ecs/components/CylinderShape.proto [hunk 4]
```diff
@@ -4,9 +4,9 @@ import "common/id.proto";
...
 
 message PBCylinderShape {  
-  bool with_collisions = 1;
-  bool is_pointer_blocker = 2;
-  bool visible = 3;
-  float radius_top = 4;
-  float radius_bottom = 5;
+  optional bool with_collisions = 1; // default=true
+  optional bool is_pointer_blocker = 2; // default=true
+  optional bool visible = 3; // default=true
```
Comment (menduz) [inline ecs/components/CylinderShape.proto:9]:
```suggestion
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool with_collisions = 1; // default=true
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool is_pointer_blocker = 2; // default=true
  // @deprecated use HiddenComponent instead https://github.com/decentraland/sdk/issues/353
  optional bool visible = 3; // default=true
```

File: ecs/components/GLTFShape.proto [hunk 5]
```diff
@@ -4,8 +4,8 @@ import "common/id.proto";
 option (ecs_component_id) = 1041;
 
 message PBGLTFShape {
-  bool with_collisions = 1;
-  bool is_pointer_blocker = 2;
-  bool visible = 3;
+  optional bool with_collisions = 1; // default=true
+  optional bool is_pointer_blocker = 2; // default=true
+  optional bool visible = 3; // default=true
```
Comment (menduz) [inline ecs/components/GLTFShape.proto:9]:
```suggestion
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool with_collisions = 1; // default=true
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool is_pointer_blocker = 2; // default=true
  // @deprecated use HiddenComponent instead https://github.com/decentraland/sdk/issues/353
  optional bool visible = 3; // default=true
```

File: ecs/components/NFTShape.proto [hunk 6]
```diff
@@ -6,11 +6,11 @@ option (ecs_component_id) = 1040;
 import "common/Color3.proto";
 
 message PBNFTShape {
-  bool with_collisions = 1;
-  bool is_pointer_blocker = 2;
-  bool visible = 3;
+  optional bool with_collisions = 1; // default=true
+  optional bool is_pointer_blocker = 2; // default=true
+  optional bool visible = 3; // default=true
```
Comment (menduz) [inline ecs/components/NFTShape.proto:11]:
```suggestion
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool with_collisions = 1; // default=true
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool is_pointer_blocker = 2; // default=true
  // @deprecated use HiddenComponent instead https://github.com/decentraland/sdk/issues/353
  optional bool visible = 3; // default=true
```

File: ecs/components/OnPointerDown.proto [hunk 7]
```diff
@@ -5,8 +5,8 @@ option (ecs_component_id) = 1060;
 import "common/ActionButton.proto";
 
 message PBOnPointerDown {
-  ActionButton button = 1;
-  string hover_text = 2;
-  float distance = 3;
-  bool show_feedback = 4;
+  optional ActionButton button = 1;
+  optional string hover_text = 2;
+  optional float distance = 3;
```
Comment (menduz) [inline ecs/components/OnPointerDown.proto:10]:
```suggestion
  optional float max_distance = 3;
```

File: ecs/components/OnPointerDownResult.proto [hunk 8]
```diff
@@ -6,12 +6,12 @@ import "common/Vector3.proto";
...
-  float distance = 8;
-  int32 timestamp = 9;
+  optional ActionButton button = 2;
+  optional string meshName = 3;
+  optional Vector3 origin = 4;
+  optional Vector3 direction = 5;
+  optional Vector3 point = 6;
+  optional Vector3 normal = 7;
+  optional float distance = 8;
+  optional int32 timestamp = 9;
```
Comment (menduz) [inline ecs/components/OnPointerDownResult.proto:16]: I think that for the result we want all values to exist

File: ecs/components/OnPointerUp.proto [hunk 9]
```diff
@@ -5,8 +5,8 @@ option (ecs_component_id) = 1061;
 import "common/ActionButton.proto";
 
 message PBOnPointerUp {
-  ActionButton button = 1;
-  string hover_text = 2;
-  float distance = 3;
-  bool show_feedback = 4;
+  optional ActionButton button = 1;
+  optional string hover_text = 2;
+  optional float distance = 3;
```
Comment (menduz) [inline ecs/components/OnPointerUp.proto:10]:
```suggestion
  optional float max_distance = 3;
```

File: ecs/components/OnPointerUpResult.proto [hunk 10]
```diff
@@ -6,12 +6,12 @@ import "common/Vector3.proto";
...
-  float distance = 8;
-  int32 timestamp = 9;
+  optional ActionButton button = 2;
+  optional string meshName = 3;
+  optional Vector3 origin = 4;
+  optional Vector3 direction = 5;
+  optional Vector3 point = 6;
+  optional Vector3 normal = 7;
+  optional float distance = 8;
+  optional int32 timestamp = 9;
```
Comment (menduz) [inline ecs/components/OnPointerUpResult.proto:16]: Same here, we will always send all values from the engine. Removing the optionals may be good

File: ecs/components/PlaneShape.proto [hunk 11]
```diff
@@ -4,8 +4,8 @@ import "common/id.proto";
 option (ecs_component_id) = 1014;
 
 message PBPlaneShape {  
-  bool with_collisions = 1;
-  bool is_pointer_blocker = 2;
-  bool visible = 3;
+  optional bool with_collisions = 1; // default=true
+  optional bool is_pointer_blocker = 2; // default=true
+  optional bool visible = 3; // default=true
```
Comment (menduz) [inline ecs/components/PlaneShape.proto:9]:
```suggestion
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool with_collisions = 1; // default=true
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool is_pointer_blocker = 2; // default=true
  // @deprecated use HiddenComponent instead https://github.com/decentraland/sdk/issues/353
  optional bool visible = 3; // default=true
```

File: ecs/components/SphereShape.proto [hunk 12]
```diff
@@ -4,7 +4,7 @@ import "common/id.proto";
 option (ecs_component_id) = 1015;
 
 message PBSphereShape {  
-  bool with_collisions = 1;
-  bool is_pointer_blocker = 2;
-  bool visible = 3;
+  optional bool with_collisions = 1; // default=true
+  optional bool is_pointer_blocker = 2; // default=true
+  optional bool visible = 3; // default=true
```
Comment (menduz) [inline ecs/components/SphereShape.proto:9]:
```suggestion
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool with_collisions = 1; // default=true
  // @deprecated use MeshCollider instead https://github.com/decentraland/sdk/issues/366
  optional bool is_pointer_blocker = 2; // default=true
  // @deprecated use HiddenComponent instead https://github.com/decentraland/sdk/issues/353
  optional bool visible = 3; // default=true
```

File: ecs/components/TextShape.proto [hunk 13]
```diff
@@ -7,27 +7,27 @@ import "common/Color3.proto";
...
-  int32 line_count = 16;
-  bool text_wrapping = 17;
-  float shadow_blur = 18;
-  float shadow_offsetX = 19;
-  float shadow_offsetY = 20;
-  float outline_width = 21;
-  Color3 shadow_color = 22;
-  Color3 outline_color = 23;
-  Color3 text_color = 24;
+  optional bool visible = 2; // default=true
```
Comment (menduz) [inline ecs/components/TextShape.proto:10]:
```suggestion
  // @deprecated use HiddenComponent instead https://github.com/decentraland/sdk/issues/353
  optional bool visible = 2; // default=true
```

**Review by menduz — DISMISSED** [review]

**Comment by menduz** [comment]
Please be mindful that this PR will break compatibility in all the scenes using the ECS7

File: ecs/components/OnPointerDown.proto [hunk 14]
```diff
@@ -5,8 +5,8 @@ option (ecs_component_id) = 1060;
 import "common/ActionButton.proto";
 
 message PBOnPointerDown {
-  ActionButton button = 1;
-  string hover_text = 2;
-  float distance = 3;
-  bool show_feedback = 4;
+  optional ActionButton button = 1; // default=ActionButton.ANY
+  optional string hover_text = 2; // default='Interact'
+  optional float max_distance = 3; // default=100
```
Comment (kuruk-mm) [inline ecs/components/OnPointerDown.proto:10]:
```suggestion
  optional float max_distance = 3; // default=10
```

File: ecs/components/OnPointerUp.proto [hunk 15]
```diff
@@ -5,8 +5,8 @@ option (ecs_component_id) = 1061;
 import "common/ActionButton.proto";
 
 message PBOnPointerUp {
-  ActionButton button = 1;
-  string hover_text = 2;
-  float distance = 3;
-  bool show_feedback = 4;
+  optional ActionButton button = 1; // default=ActionButton.ANY
+  optional string hover_text = 2; // default='Interact'
+  optional float max_distance = 3; // default=100
```
Comment (kuruk-mm) [inline ecs/components/OnPointerUp.proto:10]:
```suggestion
  optional float max_distance = 3; // default=10
```

**Review by Suduck — APPROVED** [review]
LGTM! There are some default values that are not specified in the proto. I have marked them in the Unity PR.

Please specify all the default values that are missing

**Review by nearnshaw — APPROVED** [review]

**Pull request merged**